javascript/code-challenges/challenges-06.test.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named getNames that, given an array of people objects, uses map to return an array of names reversed.
+
+For example:
+[
+{
+  name: 'lloyd',
+  age: 32,
+  shoeSize: 12
+},
+{
+  name: 'jamie',
+  age: 21,
+  shoeSize: 8
+}
+]
+
+Returns: ['dyoll', 'eimaj'];
+------------------------------------------------------------------------------------------------ */
+
+const getNames = (arr) => {
+  // Solution code here...
+  return arr.map(personObj => {
+    let strArr = personObj.name.split('');
+    return strArr.reduce((revStr, currentLetter) => {
+      return currentLetter + revStr;
+    }, '');
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function that appends ' The end.' to a string, and returns the modified string. The original source string should not be modified.
+
+------------------------------------------------------------------------------------------------ */
+
+const appendTheEnd = (str) => {
+  // Solution code here...
+  let newStr = `${str} The end.`;
+  return newStr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that accepts an array and copies the first element to the end of the array. The change should be reflected in the source array that was passed in to the function. That is, the function should modify the array 'in place'.
+
+Do not use a return statement.
+
+For example:
+const a = [1, 2, 3];
+appendFirstToLast(a);
+console.log(a) prints [1, 2, 3, 1]
+------------------------------------------------------------------------------------------------ */
+
+const appendFirstToLast = (arr) => {
+  // Solution code here...
+  return arr.forEach((element, idx) => {
+    if (idx === 0) arr.push(element);
+  });
+};
+
```

Comment:
Hi there, good work!

This challenge asked to not use a return statement, since the goal is to modify the array in place. The test in passing even though this is returning a new arr. I am thinking we need to write another test here, one that checks the value of the return from the function. 👍